### DISCUSSION TITLE
File/folder access with IStorageProvider in Android

### DISCUSSION THREAD

**Discussion by SuperJMN** [AvaloniaUI/Avalonia#12607] [Q&A]
I would like to create a File Explorer. It's working already for desktop platforms, but I can't manage file access for sandboxes devices, like Android.

How could someone possibly get access to the shared storage and external storage folders with Avalonia?

Thank you.

**Comment by timunie** [comment]
I think it's similar to what had to be done in Xamarin. Maybe you find some good guides for it there. 

- https://learn.microsoft.com/en-us/xamarin/android/app-fundamentals/permissions?tabs=windows
- https://learn.microsoft.com/en-us/xamarin/android/platform/files/
